### PR TITLE
[CAS-600] Query channels controller tests

### PR DESCRIPTION
**Pull request by ogkuzmin**
### 🎯 Goal

Cover logic of `QueryChannelsController::query`
### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (CMS, cookbooks, tutorial)
- [x] Reviewers added

### 🎉 GIF

investigator

### CODE REVIEW

File: stream-chat-android-offline/src/test/java/io/getstream/chat/android/offline/querychannels/WhenQuery.kt [hunk 1]
```diff
@@ -1,44 +1,242 @@
...
-            sut.query()
+    fun `Given DB with query channels Should invoke selectAndEnrichChannels in ChatDomain`() = runBlockingTest {
+        val chatDomainImpl = mock<ChatDomainImpl>()
+        val sut = Fixture()
+            .givenChatDomain(chatDomainImpl)
+            .givenFailedNetworkRequest()
+            .givenQueryChannelsSpec(
+                QueryChannelsSpec(
+                    Filters.neutral(),
+                    QuerySort.Companion.desc(Channel::lastMessageAt),
```
Comment (leandroBorgesFerreira) [inline stream-chat-android-offline/src/test/java/io/getstream/chat/android/offline/querychannels/WhenQuery.kt:55]:
```suggestion
                    QuerySort.desc(Channel::lastMessageAt),
```

File: stream-chat-android-offline/src/test/java/io/getstream/chat/android/offline/querychannels/WhenQuery.kt [hunk 2]
```diff
@@ -1,44 +1,242 @@
...
+
+            whenever(chatDomainImpl.channel(any<String>())) doAnswer { invocationOnMock ->
+                val cid = invocationOnMock.arguments[0] as String
+                mock<ChannelControllerImpl> {
+                    on { toChannel() } doReturn channels.first { it.cid == cid }
+                }
+            }
+            whenever(chatDomainImpl.channel(any<Channel>())) doAnswer { invocationOnMock ->
+                val channel = invocationOnMock.arguments[0] as Channel
+                mock<ChannelControllerImpl> {
```
Comment (leandroBorgesFerreira) [inline stream-chat-android-offline/src/test/java/io/getstream/chat/android/offline/querychannels/WhenQuery.kt:229]: no need for the `<ChannelControllerImpl>`. Smartcast will figure it out already. Happens in other places.

**Review by leandroBorgesFerreira — APPROVED** [review]
Just 2 comments.

File: stream-chat-android-offline/src/test/java/io/getstream/chat/android/offline/querychannels/WhenQuery.kt (hunk 2 above)
Comment (ogkuzmin) [reply stream-chat-android-offline/src/test/java/io/getstream/chat/android/offline/querychannels/WhenQuery.kt:229]: I leave it only for readability, seems it wouldn't be so easy to understand if I did omit here

**Pull request merged**